bin/cask

```diff
@@ -275,6 +294,16 @@ def exec_command(command):
     # Copy the environment and update the paths
     ENVB[b'EMACSLOADPATH'] = get_cask_path('load-path')
     ENVB[b'PATH'] = get_cask_path('path')
+
+    ## special handling for where we have a CASK_EMACS, we can use that and
+    ## pass it as the EMACS variable to the command
+    if (b'CASK_EMACS' in ENVB):
+        ENVB[b'EMACS'] = ENVB[b'CASK_EMACS']
+        ## we also remove INSIDE_EMACS because EMACS should now be
+        ## trustworthy.
+        if b'INSIDE_EMACS' in ENVB:
+            del ENVB[b'INSIDE_EMACS']
```

Comment:
I don't like this.  Even tools running within Emacs and Cask might have a valid reason to find out whether they run in Emacs.  We should preserve `$INSIDE_EMACS` under all circumstances, and I strongly disagree with dropping it here, no matter what the reason is.